front/src/common/Map/Settings/MapSettingsSignals.js

```diff
@@ -16,13 +16,38 @@ export default function MapSettingsSignals() {
     lights: lightsIcon,
   };
 
+  const switchAllSettings = (settingSwitch, stateOfSettings) => {
+    const allSettings = {};
+
+    CONSTS_SETTINGS.forEach(
+      (settingKey) => (allSettings[settingKey] = !stateOfSettings[settingSwitch])
+    );
+
+    return allSettings;
+  };
+
+  const checkedSettings = (settingSwitch, stateOfSettings) => {
+    const filteredSettingKeys = Object.keys(stateOfSettings).filter(
+      (key) => key !== settingSwitch && key !== 'all'
+    );
+    const settingValues = filteredSettingKeys.map((settingKey) => stateOfSettings[settingKey]);
+    return settingValues.every((settingValue) => settingValue === !stateOfSettings[settingSwitch]);
+  };
+
   const setSignalsList = (setting) => {
-    dispatch(
-      updateSignalsSettings({
+    let newSettings;
+
+    if (setting === 'all') {
+      newSettings = switchAllSettings(setting, signalsSettings);
```

Comment:
To go one step furthere in the refactoring, we can see that the first two blocks are doing the same thing :

`newSettings = switchAllSettings(setting, signalsSettings);`

you can regroup in one single block :

```
if (setting === 'all' || checkedSettings(setting, signalsSettings)) {
  newSettings = switchAllSettings(setting, signalsSettings);
} else { 
…
}
```